ecl/resource_ecl_provider_connectivity_tenant_connection_v2.go

```diff
@@ -0,0 +1,461 @@
+	if err != nil {
+		return CheckDeleted(d, err, "Provider Connectivity Tenant Connection")
+	}
+	log.Printf("[DEBUG] Retrieved Provider Connectivity Tenant Connection %s: %+v", tenantConnection.Name, tenantConnection)
+
+	d.SetId(tenantConnection.ID)
+	d.Set("tenant_connection_request_id", tenantConnection.TenantConnectionRequestID)
+	d.Set("name", tenantConnection.Name)
+	d.Set("description", tenantConnection.Description)
+	d.Set("tags", tenantConnection.Tags)
+	d.Set("tenant_id", tenantConnection.TenantID)
+	d.Set("name_other", tenantConnection.NameOther)
+	d.Set("description_other", tenantConnection.DescriptionOther)
+	d.Set("tags_other", tenantConnection.TagsOther)
+	d.Set("tenant_id_other", tenantConnection.TenantIDOther)
+	d.Set("network_id", tenantConnection.NetworkID)
+	d.Set("device_type", tenantConnection.DeviceType)
+	d.Set("device_id", tenantConnection.DeviceID)
+	d.Set("device_interface_id", tenantConnection.DeviceInterfaceID)
+	d.Set("port_id", tenantConnection.PortID)
+	d.Set("status", tenantConnection.Status)
+
+	return nil
+}
+
+func resourceProviderConnectivityTenantConnectionV2Update(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	var hasChange bool
+	var updateOpts tenant_connections.UpdateOpts
+
+	if d.HasChange("name") {
+		hasChange = true
+		updateOpts.Name = d.Get("name").(string)
+	}
+
+	if d.HasChange("description") {
+		hasChange = true
+		updateOpts.Description = d.Get("description").(string)
+	}
+
+	if d.HasChange("tags") {
+		hasChange = true
+		updateOpts.Tags = getTags(d, "tags")
+	}
+
+	if d.HasChange("name_other") {
+		hasChange = true
+		updateOpts.NameOther = d.Get("name_other").(string)
+	}
+
+	if d.HasChange("description_other") {
+		hasChange = true
+		updateOpts.DescriptionOther = d.Get("description_other").(string)
+	}
+
+	if d.HasChange("tags_other") {
+		hasChange = true
+		updateOpts.TagsOther = getTags(d, "tags_other")
+	}
+
+	if hasChange {
+		r := tenant_connections.Update(client, d.Id(), updateOpts)
+		if r.Err != nil {
+			return fmt.Errorf("error updating ECL Provider Connectivity Tenant Connection: %s", r.Err)
+		}
+		log.Printf("[DEBUG] Tenant Connection has successfully updated.")
+	}
+
+	return resourceProviderConnectivityTenantConnectionV2Read(d, meta)
+}
+
+func resourceProviderConnectivityTenantConnectionV2Delete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	tenantConnection, err := tenant_connections.Get(client, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "Provider Connectivity Tenant Connection")
+	}
+
+	deviceType := tenantConnection.DeviceType
```

Comment:
I think you not need to define variable here because it is used just once.

Reply:
that's right.